### PR TITLE
Document process for proposing changes to SIG/WG/UG leadership

### PR DESCRIPTION
**Pull request by nikhita**
Fixes https://github.com/kubernetes/steering/issues/179

This PR adds a light-weight process for proposing changes to SIG/WG/UG leadership. This will help in creating a consistent process for feedback from the community around leadership changes.

/hold
for review + steering approval

cc @kubernetes/steering-committee

### CODE REVIEW

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (liggitt) [inline contributors/chairs-and-techleads/leadership-changes.md:22]: seems like the mechanics should come _before_ updating sigs.yaml:

* org member
* completed required training

**Review by liggitt — COMMENTED** [review]
thanks for putting together a checklist for this. just had a couple comments

**Review by mrbobbytables — COMMENTED** [review]
I think its a good summary of what's generally done to start. A follow-up to do either in this doc or another one is outline what needs to be done to rotate out of all the various communication properties.

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (dims) [reply contributors/chairs-and-techleads/leadership-changes.md:15]: if we are looking for patterns else where :) Yes, 3 business days or 72 hours sounds good

```PMCs SHOULD ensure that decision-making processes allow input for a sufficient amount of time - typically at least 72 hours - so that project participants in various time zones have a chance to participate in the decision.``` (from https://www.apache.org/dev/pmc.html)

Comment (derekwaynecarr) [reply contributors/chairs-and-techleads/leadership-changes.md:15]: Given that most SIGs meet on a weekly or bi-weekly cadence, should we just settle on 1w?

Comment (dims) [reply contributors/chairs-and-techleads/leadership-changes.md:15]: +1 to 1w

Comment (derekwaynecarr) [reply contributors/chairs-and-techleads/leadership-changes.md:15]: An issue with lazy consensus is it requires active objection to know it was not reached.  If concerns are shared privately with existing leads (or separately to steering-private), we may want to provide additional buffer beyond 72 hours to ensure any communication is handled and concerns are satisfied.  I think a week is probably sufficient.

Comment (liggitt) [reply contributors/chairs-and-techleads/leadership-changes.md:15]: if concerns are raised to the current leads or to steering, it would make sense for the leads or steering to `/hold` the PR until those are resolved.

Comment (mrbobbytables) [reply contributors/chairs-and-techleads/leadership-changes.md:15]: I think 1 week is an acceptable timeframe for leadership changes. It's an infrequent and large enough event to warrant a longer feedback period.

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (nikhita) [reply contributors/chairs-and-techleads/leadership-changes.md]: > completed required training

We ask leads to complete the training within 30 days of their appointment but I added it as a prerequisite here to try enforcing it as a best practice 

https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#all-leads

**Comment by nikhita** [comment]
Addressed feedback. PTAL.

**Comment by mrbobbytables** [comment]
/lgtm
Thanks @nikhita :)

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (ehashman) [inline contributors/chairs-and-techleads/leadership-changes.md:25]: I feel like there's a *lot* of updates required in the org repo for leads, not just the OWNERS_ALIASES. Using myself as a reference, I see that there's also teams.yaml files that need updates in the kubernetes-sigs and kubernetes configs, as well as the milestone-maintainers team under the release team config.

I was adding myself to OWNERS files and teams months after becoming a lead so it might be good to document them all at once :sweat_smile: I think @mrbobbytables mentioned this in his review.

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (ehashman) [inline contributors/chairs-and-techleads/leadership-changes.md:22]: My SIG was unable to reach lazy consensus because multiple leads ran for an open position, so we had an election using CIVS with voters defined from the SIG membership defined in kubernetes/org. Should we include some possible options for when lazy consensus isn't achieved?

**Comment by dims** [comment]
@nikhita let's add a bit more per @ehashman 's comments. LGTM otherwise

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (nikhita) [reply contributors/chairs-and-techleads/leadership-changes.md:22]: > Should we include some possible options for when lazy consensus isn't achieved?

That's a good point!

@mrbobbytables do you know what process was followed for the SIG lead election in this case? I know contribex was involved, but I can't recall the details.

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (nikhita) [reply contributors/chairs-and-techleads/leadership-changes.md]: > teams.yaml files that need updates in the kubernetes-sigs and kubernetes configs, as well as the milestone-maintainers team under the release team config.

Done :+1:

**Review by irvifa — COMMENTED** [review]
/lgtm

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (mrbobbytables) [reply contributors/chairs-and-techleads/leadership-changes.md:22]: ContribEx was consulted, but was not involved in the execution IIRC. We do have an oustanding item to document the process. - TL;DR -  ContribEx will handle the execution for a CIVS election if provided a list of emails and candidates as a neutral party.

Comment (ehashman) [reply contributors/chairs-and-techleads/leadership-changes.md:22]: Since https://github.com/kubernetes/community/issues/5096 is in flight I'm hoping we can avoid direct references/codifying CIVS -- maybe we can keep this general and just indicate that if there isn't lazy consensus,

- an election should be held
- the eligible voters will be the members of the SIG as defined in kubernetes/org
- contribex should be contacted to assist in the administration of the election

**Comment by justaugustus** [comment]
@ehashman -- for some additional bits and maybe some inspiration (since SIG Release is [transitioning some roles](https://github.com/kubernetes/community/pull/5286) as well)...

SIG Release has a few issue templates that could be helpful:
- Practical example for Tim (which has this PR's content stuck on top): https://github.com/kubernetes/sig-release/issues/1321
- [SIG Release Chair/TL template](https://github.com/kubernetes/sig-release/issues/new?labels=sig%2Frelease%2C+area%2Frelease-eng&template=lead.md&title=Chair%2FTechnical+Lead+access+for+%3CGH-handle%3E)
- [Release Manager template](https://github.com/kubernetes/sig-release/issues/new?labels=sig%2Frelease%2C+area%2Frelease-eng&template=release-manager.md&title=Release+Manager+access+for+%3CGH-handle%3E)
- [Release Team Lead template](https://github.com/kubernetes/sig-release/issues/new?labels=sig%2Frelease%2C+area%2Frelease-eng%2C+area%2Frelease-team&template=release-team-lead.md&title=Release+Team+Lead+access+for+%3CGH-handle%3E)

**Comment by justaugustus** [comment]
/lgtm
(as a Chair)

Thanks for adding this!

File: contributors/chairs-and-techleads/leadership-changes.md
Comment (nikhita) [reply contributors/chairs-and-techleads/leadership-changes.md:22]: Thanks @mrbobbytables and @ehashman!

Added a note about elections. We can add detailed steps/checklist for the election once we figure out the process: https://github.com/kubernetes/community/issues/4586

**Review by ehashman — APPROVED** [review]
/lgtm

**Comment by nikhita** [comment]
Going to cancel the hold since we have many LGTMs. Let's merge and iterate if further changes are needed.

/hold cancel

**Pull request merged**
